crates/oxc_transformer/src/plugins/styled_components.rs

```diff
@@ -911,11 +914,10 @@ fn minify_template_literal<'a>(lit: &mut TemplateLiteral<'a>, ast: AstBuilder<'a
                 // //         ^^^^^^^^^^^^^^^^^^^^^^
                 // ```
                 quasis[quasi_index - 1].span.end = quasi.span.end;
-
-                // Remove this quasi and the preceding expression.
-                // TODO: Remove scopes, symbols, and references for removed `Expression`.
-                quasis.remove(quasi_index);
-                expressions.remove(quasi_index - 1);
+                // Mark previous quasi for deletion, it will be deleted all of quasis are processed.
+                quasis[quasi_index - 1].value.raw = QUASI_DELETION_MARKER;
```

Comment:
It's *this* quasi which should be deleted, not the previous one.

Reply:
Weird that all tests still passing with this mistake. Don't the tests cover this? Or am I mistaken?

Reply:
I've got it now. I'll update the comments in a follow-on PR to reflect this change in the logic.